rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -767,20 +768,23 @@ protected void doCODESIZE() {
             spendOpCodeGas();
         }
         // EXECUTION PHASE
-        int length;
-        if (op == OpCode.CODESIZE)
-            //TODO(mmarquez): we need to add support to precompiled contracts
-        {
-            length = program.getCode().length; // during initialization it will return the initialization code size
+        DataWord codeLength;
+        if (op == OpCode.CODESIZE) {
+            codeLength = new DataWord(program.getCode().length); // during initialization it will return the initialization code size
         } else {
             DataWord address = program.stackPop();
-            length = program.getCodeAt(address).length;
+            codeLength = new DataWord(program.getCodeAt(address).length);
+            if (program.getBlockchainConfig().isRfs90()) {
```

Comment:
`isRfs50`?

Reply:
You are right! thanks for checking this!